actionview/lib/action_view/helpers/tag_helper.rb

```diff
@@ -400,6 +399,11 @@ def escape_once(html)
       end
 
       private
+        def ensure_valid_html5_tag_name(name)
+          raise ArgumentError, "Invalid HTML5 tag name: #{name}" unless /\A[a-zA-Z][^\s\/>]*\z/.match?(name)
```

Comment:
Probably should use `"#{name.inspect}` here to make the error message readable if there's an empty string or embedded strings
```suggestion
          raise ArgumentError, "Invalid HTML5 tag name: #{name.inspect}" unless /\A[a-zA-Z][^\s\/>]*\z/.match?(name)
```